### ISSUE TITLE
Ability to disable Rollout.io from the SDK

### ISSUE THREAD

**Issue by ekeren** [rollout/rollout.io-ios#23]
Once Rollout.io is setup there is no way for the application to force disable it from the SDK.
Only when the application is loaded the application can choose not to call setup function, but this does not allow to disable it even if it was already loaded.

This is important for some users using feature toggling and want to be able to opt-out Rollout.io SDK.

**Comment by x2on** [comment]
:thumbsup: please add this - actually this is a show stopper for me

**Comment by ekeren** [comment]
@x2on thanks for the request. Currently once you remove all the hot-patches from the Rollout.io dashboard the SDK will only do minimal network work and find out that there is nothing to patch. So basically when there aren't any patches on Rollout.io dashboard it doesn't do anything to your code. 

If this is not enought you can decide to opt in/out Rollout.io SDK setup, something like this: 

```objc
if ([MyFeatureToggleSystem isRolloutEnabled]){
  [Rollout setupWithDebug:YES];
}
```
in ```application:didFinishLuanchingWithOptions:```. 

if you do not call the setup function on Rollout it __will not__ load anything to your runtime, we do not do run any ```+load``` or ```+initialize``` functions from the SDK

This technique is usually used by our users to make sure rollout doesn't affect their applications performance indicators (rolling out  Rollout.io). We encourage users to try us in small numbers but then to enable Rollout.io to all your users so it will be easiyer to patch your app.

The disadvantage of toggling Rollout setup code that it only happens on application startup. 
This ticket talks about the ability to disable Rollout.io during runtime, disable that will not require a restart to take place

After understanding this, is this feature still a showstopper for you?

**Comment by x2on** [comment]
@ekeren yes, because we have the policy that we need the possibility to deactivate every third party SDK remotely at any time. With your method this could only be done at a complete clean app start.

**Comment by ekeren** [comment]
@x2on right.
Let me think consult my team regarding this, we might push it up the priority queue a bit.

**Comment by silyevsk** [comment]
We've added this in 0.9 (which is going to be released soon).

Here is how you can enable/disable snippets to enable/disable the SDK:
* To disable or enable the SDK:

  `[Rollout setRolloutDisabled:YES]; // disable the SDK`
  `[Rollout setRolloutDisabled:NO]; // enable the SDK`
* To check the current status:

  `[Rollout rolloutDisabled] // returns YES if disabled`
* To start the SDK as disabled:

  `RolloutOptions *options = [[RolloutOptions alloc] init];`
  `options.rolloutDisabled = YES;`
  `[Rollout setupWithDebug:<#(BOOL)#> options:options];`

**Comment by ekeren** [comment]
@glencoates, this feature allows you to disable a Rollout.io at any moment from your own code. FYI

**Comment by glencoates** [comment]
Thanks for the heads up :)

**Comment by ekeren** [comment]
issue resolved at 0.9
